core/src/main/scala/kafka/tools/StreamsResetter.java

```diff
@@ -219,23 +240,85 @@ private void resetInputAndInternalAndSeekToEndIntermediateTopicOffsets() {
                 }
             }
 
-            if (inputAndInternalTopicPartitions.size() > 0) {
-                client.seekToBeginning(inputAndInternalTopicPartitions);
+            maybeSeekToBeginning(client, inputAndInternalTopicPartitions, internalTopics);
+
+            maybeSeekToEnd(client, intermediateTopicPartitions);
+
+            if (!dryRun) {
+                for (final TopicPartition p : partitions) {
+                    client.position(p);
+                }
+                client.commitSync();
             }
-            if (intermediateTopicPartitions.size() > 0) {
-                client.seekToEnd(intermediateTopicPartitions);
+
+            if (notFoundInputTopics.size() > 0) {
+                System.out.println("Following input topics are not found, skipping them");
+                for (String topic : notFoundInputTopics) {
+                    System.out.println("Topic - " + topic);
+                }
             }
 
-            for (final TopicPartition p : partitions) {
-                client.position(p);
+            if (notFoundIntermediateTopics.size() > 0) {
+                System.out.println("Following intermediate topics are not found, skipping them");
+                for (String topic : notFoundIntermediateTopics) {
+                    System.out.println("Topic - " + topic);
+                }
             }
-            client.commitSync();
+            client.close();
```

Comment:
Not required. Client will be automatically closes, as we use Java's "try with resource" feature.